deepsensor/data/loader.py

```diff
@@ -853,31 +853,26 @@ def _compute_x1x2_direction(self) -> str:
                 coord_x1_right= var.x1[-1]
                 coord_x2_top= var.x2[0]
                 coord_x2_bottom= var.x2[-1]
-            #Todo- what to input for pd.dataframe
-            elif isinstance(var, (pd.DataFrame, pd.Series)):
-                var_x1_min = var.index.get_level_values("x1").min()
-                var_x1_max = var.index.get_level_values("x1").max()
-                var_x2_min = var.index.get_level_values("x2").min()
-                var_x2_max = var.index.get_level_values("x2").max()
+            
+                x1_ascend = True if coord_x1_left <= coord_x1_right else False
+                x2_ascend = True if coord_x2_top <= coord_x2_bottom else False
 
-            x1_ascend = True
-            x2_ascend = True
-            if coord_x1_left < coord_x1_right:
-                x1_ascend = True
-            if coord_x1_left > coord_x1_right:
-                x1_ascend = False
+                coord_directions = {
+                        "x1": x1_ascend,
+                        "x2": x2_ascend,
+                    }
 
-            if coord_x2_top < coord_x2_bottom:
-                x2_ascend = True
-            if coord_x2_top > coord_x2_bottom:
-                x2_ascend = False
-
-
-
-        coord_directions = {
-                "x1": x1_ascend,
-                "x2": x2_ascend,
-            }
+            #TODO- what to input for pd.dataframe
+            elif isinstance(var, (pd.DataFrame, pd.Series)):
+                # var_x1_min = var.index.get_level_values("x1").min()
+                # var_x1_max = var.index.get_level_values("x1").max()
+                # var_x2_min = var.index.get_level_values("x2").min()
+                # var_x2_max = var.index.get_level_values("x2").max()
+
+                coord_directions = {
+                    "x1": None,
+                    "x2": None
+                }
```

Comment:
+ Check the direction for both context and target, return if they're consistent and raise error if not?
+ Nones for pandas data

Reply:
Looking into this:
+ `_compute_x1x2_direction` is currently called in `TaskLoader.__init__()`,
+ The results from this method are only used in `TaskLoader.sample_sliding_window()`,
+ When we last talked about this @MartinSJRogers - I think we thought that `_compute_x1x2_direction` only needed to work for xarray data (not pandas), but in `test_sliding_window` that Nils' wrote, pandas data is passed in - which causes errors in executing these functions.

Do these functions, in fact, need to work for pandas data? Can you help clear this up?
Is the pandas case actually covered by the bottom `else` case here:
https://github.com/davidwilby/deepsensor/blob/79afa1263f392cc8f2a86221f144c18fcaddb97f/deepsensor/data/loader.py#L1481-L1543

Also, I think we should rename the `coord_directions` variable to `coord_ascending` or `coord_ascending_from_top_left` if we're going to keep it as a dictionary of booleans - so that it is more clear what the booleans indicate.

Reply:
Agree with the renaming. I think `coord_ascending_from_top_left` is the best as it is the most explicit. Please see additional suggested changes in my msjr/stitch pull request as I think this addresses this problem at last!

Reply:
Since you've partially done this in #6 I'll go ahead and merge this PR so we're not trying to make the same changes on multiple branches.